jsonschema/cli.py

```diff
@@ -69,19 +70,49 @@ def validation_success(self, **kwargs):
 
 @attr.s
 class _PrettyFormatter(object):
+    _MESSAGE_BAR_CHAR = '═'
+    _MESSAGE_CORNER_CHARS = ('╒', '╕')
+    _MESSAGE_FORMAT = '{}══[{}]═══({})'
+    _MESSAGE_MAX_LENGTH = 79
+
+    @classmethod
+    def _json_formatter(cls, x):
+        return json.dumps(x, separators=(',\n', ': '), sort_keys=True)
 
-    _ERROR_MSG = dedent(
-        """\
-        ===[{type}]===({path})===
+    def _message_end_chars(self, header=False):
+        return self._MESSAGE_CORNER_CHARS if header is True else [self._MESSAGE_BAR_CHAR] * 2
 
-        {body}
-        -----------------------------
-        """,
-    )
-    _SUCCESS_MSG = "===[SUCCESS]===({path})===\n"
+    def _message_line(self, path, type, header=False):
+        begin_char, end_char = self._message_end_chars(header)
+        return self._MESSAGE_FORMAT.format(begin_char, type, path) \
+            .ljust(self._MESSAGE_MAX_LENGTH - 1, self._MESSAGE_BAR_CHAR) + end_char
+
+    if len(_MESSAGE_BAR_CHAR) != 1:
+        # The code in this if-block is for Python interpreters that don't
+        # treat multibyte Unicode characters as single characters.
+        # E.g., some versions of Python 2.x.  This block may be removed
+        # when support for those interpreters is no longer needed.
+
+        _FORMAT_LENGTH = len(
+            _MESSAGE_FORMAT.replace(_MESSAGE_BAR_CHAR, '.').format('.', '', '')) + 1
+
+        def _message_line(self, path, type, header=False):
+            begin_char, end_char = self._message_end_chars(header)
+
+            bar_length = self._MESSAGE_MAX_LENGTH - len(type) - len(path) - self._FORMAT_LENGTH
+
+            return self._MESSAGE_FORMAT.format(begin_char, type, path) + \
+                self._MESSAGE_BAR_CHAR * bar_length + end_char
+
+
+    def _error_msg(self, path, type, body):
+        HEADER = self._message_line(path, type, header=True)
+        FOOTER = '└' + '─' * (self._MESSAGE_MAX_LENGTH - 2) + '┘'
+
+        return '\n'.join((HEADER, str(body), FOOTER, '\n'))
```

Comment:
What type is body here? The `str` call looks suspicious

Reply:
I think I had that there to address a problem with Py2.  I've removed it.